sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DropTableExec.scala

```diff
@@ -26,13 +27,16 @@ import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
  * Physical plan node for dropping a table.
  */
 case class DropTableExec(
+    session: SparkSession,
     catalog: TableCatalog,
     ident: Identifier,
     ifExists: Boolean,
     purge: Boolean) extends V2CommandExec {
 
   override def run(): Seq[InternalRow] = {
     if (catalog.tableExists(ident)) {
+      val table = catalog.loadTable(ident)
+      session.sharedState.cacheManager.uncacheV2Table(table)
```

Comment:
how about `session.sharedState.cacheManager.uncacheQuery(session.table(identifer), cascade = true)`

Reply:
or more efficiently
```
val relation = DataSourceV2Relation.create(table, Some(catalog), Some(ident))
session.sharedState.cacheManager.uncacheQuery(relation, cascade = true)
```
I don't think we can cache the streaming table yet.

Reply:
Thanks @cloud-fan . Originally I used the first option but since this is called irrespective whether the target table is cached or not, it means we need to handle streaming tables as well. And since `SparkSession.table` always create `UnresolvedRelation` with `isStreaming = false`, the call will fail later on with those.

I think the second option is promising. Will try that. Thanks.